template/src/server.js

```diff
@@ -0,0 +1,69 @@
+import { createServer, Model, Response } from 'miragejs';
+
+export function makeServer({ environment = 'test' } = {}) {
+  let server = createServer({
+    environment,
+    models: {
+      users: Model,
+    },
+    seeds(server) {
+      server.create('user', {
+        username: 't10',
+        password: '12345678',
+      });
+    },
+    routes() {
+      this.namespace = 'api';
+
+      this.get('/users', (schema, request) => {
+        return schema.users.all();
+      });
+
+      this.get('/users/:id', (schema, request) => {
+        const user = schema.users.find(id);
+        if (user) {
+          return user;
+        }
+        return new Response(404, { errors: ['User not found.'] });
+      });
+
+      this.patch('/users/:id', (schema, request) => {
+        const userId = request.params.id;
+        const newUser = JSON.parse(request.requestBody);
+        const user = schema.users.find(userId);
+        if (user) {
+          return user.update(newUser);
+        }
+        return new Response(404, { errors: ['User not found.'] });
+      });
+
+      this.post('/login', (schema, request) => {
+        let reqData = JSON.parse(request.requestBody);
+        const user = schema.users.where(
+          (item) =>
+            item.username === reqData.username &&
+            item.password === reqData.password
+        );
+        if (user.models?.length) {
+          return { token: 'string' };
+        }
+        return new Response(401, {
+          errors: ['Either user or password are incorrects.'],
+        });
+      });
+
+      this.post('/reset-password/:username', (schema, request) => {
+        const user = schema.users.where(
+          (item) => item.username === request.params.username
+        );
+        if (user) {
+          const newPass = Math.random().toString(36).substring(2, 15);
+          user.update({ password: newPass });
+          return new Response(200, { data: 'Password updated.' });
+        }
+        return new Response(404, { errors: ['User not found.'] });
+      });
+    },
+  });
+  return server;
+}
```

Comment:
This is a .js file in a TS first project. Mirage does not support ts?

Reply:
I searched a little bit and didn't find out, I'll search a little bit more, I let this as a .js file because the MirageJS introduction files use it like this but I agree with your point.

Reply:
Particularly, I don't care if the mirage server is written in TS or JS because in the future it will be removed from the application.

Reply:
I found just [this unofficial TS](https://github.com/zoltan-nz/miragejs/blob/master/types/index.d.ts) files for MirageJS, and still some people that are using it are having some problems. I think it's ok let this as a .js file at least for now.

Reply:
yeah, if there is no official support, there is no point to use it.